### ISSUE TITLE
Pass the environment PATH into SConstruct for the samples

### ISSUE THREAD

**Issue by bryanwweber** [Cantera/cantera#626]
Building the samples with a compiler not located in the OS default `PATH` is not possible at the moment with SCons because SCons does not include environment variables from the shell in its `Environment`. I'm encountering this issue with the conda package, where the conda compilers should be used to compile the samples (since libcantera was also compiled with those compilers). However, the conda compilers just get added to the beginning of the `PATH`, so SCons can't find them.

I think there are two solutions here. One is to add

```python
import os
env = Environment(ENV={"PATH": os.environ["PATH"], @tmpl_env_args@)
```
to the `samples/SConstruct.in`. The other is to add a configuration variable into the `samples/SConscript` that adds the `PATH` to `@tmpl_env_args`.

I tried specifying the full path to the compiler as the `CC` and `CXX` variables, but unfortunately, conda doesn't detect them as "hard-coded" paths and do its magic to replace them. This is arguably a bug in conda; it only checks for the host environment, not the build environment, but the compilers are in the build environment).

**Comment by bryanwweber** [comment]
If this is not a change we want to make here, I can patch the template file during the conda build process instead

**Comment by speth** [comment]
Is this superseded/resolved by #648?

**Comment by bryanwweber** [comment]
Yes? #648 is one solution to this problem, by specifying the compiler name that SCons knows what to do with (i.e. `clang` on macOS, `gcc` on Linux) it seems to work. I'd say that's probably the better solution for now and if it is a problem in the future, we can revisit. I'll leave this open until we push the first alpha conda packages to the dev channel and we can actually test what happens though.

**Comment by bryanwweber** [comment]
I'm going to close this, I think it was sufficiently resolved by #648
